src/v/security/jwt.h

```diff
@@ -0,0 +1,623 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+#pragma once
```

Comment:
What is the advantage of putting everything in a header file for this?

Reply:
+1

Reply:
i guess cause it's all templates?

Reply:
but i suppose we could just instantiate them in a .cc file. surely they are generic outside of security uses.

Reply:
> What is the advantage of putting everything in a header file for this?

This is going to be included from:
* test/jwt_test.cc
* oidc_authenticator.cc

I can refactor it, but I suspect that the larger includes for json and cryptopp will remain in the header, if the bits are to be unit tested.